src/main/java/com/iexec/common/chain/eip712/EIP712Challenge.java

```
@@ -0,0 +1,86 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.common.chain.eip712;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.iexec.common.utils.HashUtils;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.*;
+
+/**
+ * EIP-712 compliant challenge based on {@link Challenge} type and {@link EIP712Domain}.
+ * <p>
+ * In the current implementation, {@link EIP712Domain} fields are {@code name}, {@code version} and {@code chainID}.
+ * There is no {@code verifyingContract}.
+ */
+@Slf4j
+@NoArgsConstructor
+public class EIP712Challenge extends EIP712Entity<Challenge> {
+
+    public EIP712Challenge(EIP712Domain domain, Challenge message) {
+        super(domain, message);
+    }
+
+    public String getPrimaryType() {
+        return "Challenge";
+    }
+
+    @Override
+    public List<TypeParam> getMessageTypeParams() {
+        return Collections.singletonList(
+                new TypeParam("challenge", "string")
+        );
+    }
+
+    @Override
+    public String getMessageHash() {
+        return super.hashMessageValues(
+                getMessage().getChallenge()
+        );
+    }
+
+    /**
+     * Represents the instance as a JSON string that will be exchanged between services.
+     * @return the JSON string representing the instance
+     */
+    public String toJsonString() {
+        ObjectMapper mapper = new ObjectMapper();
+        Arrays.asList(NON_NULL, NON_EMPTY, NON_DEFAULT).forEach(mapper::setSerializationInclusion);
+        try {
+            return mapper.writeValueAsString(this);
+        } catch (JsonProcessingException e) {
+            log.error("EIP-712 challenge serialization failed.", e);
+            return "";
```

Shouldn't this re-throw the exception instead of returning an empty String?

---

better, the method could simply be removed.
I anticipated I would need it, but that wasn't the case.
Removed in 4a6eff7

---

Thanks!